test/e2e-tests.sh

```diff
@@ -60,6 +61,20 @@ go_test_e2e -timeout=30m \
   ${parallelism} \
   "--resolvabledomain=$(use_resolvable_domain)" "${use_https}" "$(ingress_class)" || failed=1
 
+# Certificate conformance tests must be run separately
+# because they need cert-manager specific configurations.
+kubectl apply -f ./test/config/autotls/certmanager/selfsigned/
+add_trap "kubectl delete -f ./test/config/autotls/certmanager/selfsigned/ --ignore-not-found" SIGKILL SIGTERM SIGQUIT
```

Comment:
The amount of careful sequencing we're doing to install and uninstall things feels fragile and like something we're bound to regret.  cc @chaodaiG What would it take for us to have separate e2e test legs in Serving for these configurations?  I'd be much more comfortable testing with isolated clusters than potentially forming a blindspot because something wasn't properly cleaned up. 😬